elytron/WFCORE-5178_native-support-for-oidc.adoc

```diff
@@ -0,0 +1,164 @@
+greatly simplify other cases where OIDC is used (e.g., bootable JAR).
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFCORE-5178[WFCORE-5178]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/ELY-2034[ELY-2034]
+* https://issues.redhat.com/browse/WFLY-14017[WFLY-14017]
+* https://issues.redhat.com/browse/EAP7-1540[EAP7-1540]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* TBD
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE.
+// Discuss with QE during the Kickoff state to decide this
+
+[ ] Engineering
+
+[ ] QE
+
+* TBD
+
+=== Affected Projects or Components
+
+* WildFly Elytron
+* WildFly Core
+* WildFly
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+
+==== HTTP mechanism implementation
+
+A new `HttpServerAuthenticationMechanism` implementation, similar to `org.keycloak.adapters.elytron.KeycloakHttpServerAuthenticationMechanism`,
+will be added to WildFly Elytron to provide support for OIDC, as specified in the https://openid.net/specs/openid-connect-core-1_0.html[OpenID Connect Core 1.0 spec].
+
+Like the existing Keycloak client adapter, this new `HttpServerAuthenticationMechanism` implementation will only
+support the https://openid.net/specs/openid-connect-core-1_0.html#CodeFlowAuth[Authorization Code Flow] and
+Authorization Code Flow with https://tools.ietf.org/html/rfc7636[Proof Key for Code Exchange] (PKCE).
+
+==== Token storage
+
+The new HTTP mechanism implementation will be able to make use of a session token store and a cookie token store
+to store account info (i.e., access tokens, ID tokens, and refresh tokens).
+
+The session token store will be used by default but it should be possible to specify that the cookie token store
+should be used instead.
+
+The new HTTP mechanism implementation will refresh the access token when it is expired. Since the Keycloak client adapter
+provides an option to refresh the token on every request, we should provide this option as well but make
+sure to document that the use of this option is discouraged since it can have a significant impact on performance.
+
+==== New elytron-oidc subsystem
+
+A new subsystem, `elytron-oidc` will be added to WildFly. Its schema will be similar to the existing
+https://github.com/keycloak/keycloak/blob/master/adapters/oidc/wildfly/wildfly-subsystem/src/main/resources/schema/wildfly-keycloak_1_1.xsd[keycloak-adapter-subsystem].
+The new subsystem will install a `DeploymentUnitProcessor` which will detect if OIDC is required for a deployment.
+
+Like with the existing Keycloak client adapter, the OIDC configuration will be specified via the deployment
+or via subsystem configuration. For the deployment case, the OIDC configuration will be specified in a JSON
+configuration file in the WEB-INF directory along with the `auth-method`, `OIDC`, specified in the `web.xml`.
```

Comment:
Is possible use server variables in configuration of OIDC?

Reply:
Yes, it will be possible to configure this via the subsystem directly.

Reply:
Maybe the questions was whether it will be possible to use some kind of expressions even in `oidc.json`.

Reply:
Ah, that makes sense. Yes, it should be possible to use expressions in `oidc.json`.